runtime/src/bank.rs

```diff
@@ -5165,6 +5166,85 @@ impl Bank {
         self.collect_rent_eagerly(true);
     }
 
+    pub fn minimize_bank_for_snapshot(&self, mut minimized_account_set: HashSet<Pubkey>) {
+        self.minimization_add_owner_accounts(&mut minimized_account_set);
+        self.minimization_add_programdata_accounts(&mut minimized_account_set);
+        self.accounts()
+            .accounts_db
+            .minimize_accounts_db(self.slot(), &minimized_account_set);
+    }
+
+    // Used to get owner accounts in `minimize_bank_for_snapshot`
+    fn minimization_add_owner_accounts(&self, minimized_account_set: &mut HashSet<Pubkey>) {
+        let owner_accounts: HashSet<_> = self
+            .accounts()
+            .accounts_db
+            .thread_pool_clean
+            .install(|| {
+                minimized_account_set
+                    .par_iter()
+                    .filter_map(|pubkey| self.get_account(pubkey))
+                    .map(|account| *account.owner())
+            })
+            .collect();
+        self.accounts().accounts_db.thread_pool_clean.install(|| {
+            minimized_account_set.par_extend(owner_accounts);
+        });
+    }
+
+    // Used to get owner accounts in `minimize_bank_for_snapshot`
+    fn minimization_add_programdata_accounts(&self, minimized_account_set: &mut HashSet<Pubkey>) {
+        let programdata_accounts: HashSet<_> = self
+            .accounts()
+            .accounts_db
+            .thread_pool_clean
+            .install(|| {
+                minimized_account_set.par_iter().filter_map(|pubkey| {
+                    self.get_account(pubkey)
+                        .filter(|account| account.executable())
+                        .filter(|account| bpf_loader_upgradeable::check_id(account.owner()))
+                        .and_then(|account| {
+                            if let Ok(UpgradeableLoaderState::Program {
+                                programdata_address,
+                            }) = account.state()
+                            {
+                                Some(programdata_address)
+                            } else {
+                                None
+                            }
+                        })
+                })
+            })
+            .collect();
+        self.accounts()
+            .accounts_db
+            .thread_pool_clean
+            .install(|| minimized_account_set.par_extend(programdata_accounts));
+    }
+
+    pub fn get_rent_collection_accounts_between_slots(
```

Comment:
this is already in master, right?
you can rebase this pr to master to get rid of this, I think.

Reply:
The partition generation was in #25669, but the code for actual rent collection does a bit more than just getting the pubkeys. So this function is *only* getting the pubkeys, since that's all we need here.